scipy/integrate/_cubature.py

```diff
@@ -0,0 +1,617 @@
+import math
+import heapq
+import itertools
+import functools
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+import numpy as np
+
+from numpy.polynomial.polynomial import Polynomial
+
+
+def cub(f, a, b, rule, rtol=1e-05, atol=1e-08, limit=10000, args=()):
```

Comment:
what is `limit`, the number of subdivisions? A more descriptive name would be nice. Also the default, why this specific number